app/backend/approaches/chatreadretrieveread.py

```diff
@@ -236,7 +257,8 @@ async def run_without_streaming(
         extra_info, chat_coroutine = await self.run_until_final_call(
             history, overrides, auth_claims, should_stream=False
         )
-        chat_resp = dict(await chat_coroutine)
+        chat_completion_response: ChatCompletion = await chat_coroutine
+        chat_resp = chat_completion_response.model_dump()  # Convert to dict to make it JSON serializable
         chat_resp["choices"][0]["context"] = extra_info
         if overrides.get("suggest_followup_questions"):
             content, followup_questions = self.extract_followup_questions(chat_resp["choices"][0]["message"]["content"])
```

Comment:
I looked at typing the chat completion chunks that this coroutine yields, but Choice doesn't have a context or session_state parameter.

Reply:
Ah that's because context and session_state are additions that we've defined in our Azure Chat App protocol. I don't know if it'd make sense to inherit from the openai Completion object to add those?

Reply:
I'd have to define a bunch of custom models, which would remove the value of validating against OpenAI's API and make it kinda pointless.